Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -483,15 +485,82 @@ - (void)configureViews
             addObject:[bodyMediaView autoSetDimension:ALDimensionHeight toSize:bodyMediaSize.CGSizeValue.height]];
     }
 
-    [self updateBubbleColor];
+    [self insertContactShareButtonsIfNecessary];
 
-    // If we're stroking the bubble edge, ensure the stroke
-    // view is in front of its peers to prevent it from being occluded.
-    [self.bubbleStrokeView.superview bringSubviewToFront:self.bubbleStrokeView];
+    [self updateBubbleColor];
 
     [self configureBubbleRounding];
 }
 
+- (void)insertContactShareButtonsIfNecessary
+{
+    if (self.cellType != OWSMessageCellType_ContactShare) {
+        return;
+    }
+
+    if (![OWSContactShareButtonsView hasAnyButton:self.viewItem.contactShare]) {
+        return;
+    }
+
+    OWSAssert(self.viewItem.contactShare);
+
+    OWSContactShareButtonsView *buttonsView =
+        [[OWSContactShareButtonsView alloc] initWithContactShare:self.viewItem.contactShare delegate:self];
+    [buttonsView createContents];
```

Comment:
Why don't we create contents in the view's initializer?

Reply:
I don't see where we ever insert this `buttonsView`. Am I crazy?

Reply:
> Why don't we create contents in the view's initializer?

Done.  We have a _lot_ of the views involved in the message rendering now, and many of them can be used for measurement, rendering or both.  In the measurement case, perf matters, so we often deliberately avoid creating the view contents (which can involve expensive loads).  So I've been following that pattern in general, although it wasn't needed in this case.

> I don't see where we ever insert this buttonsView. Am I crazy?

See `[clipView addSubview:buttonsView]` below.